src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -62,23 +89,16 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 				msg.getRequestHeader().isSecure()){ //No point reporting for non-SSL resources
 			//Content available via both HTTPS and HTTP is a separate though related issue
 			Vector<String> STSOption = msg.getResponseHeader().getHeaders("Strict-Transport-Security");
-			if (STSOption == null) { // Header NOT found
-				Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_MEDIUM, //PluginID, Risk, Reliability
-						getName()); //Name
-			    		alert.setDetail(
-			    			getDescription(), //Description
-			    			msg.getRequestHeader().getURI().toString(), //URI
-			    			"",	// Param
-			    			"", // Attack
-			    			"", // Other info
-			    			getSolution(), //Solution
-			    			getReference(), //References
-			    			"",	// Evidence
-							16, // CWE-16: Configuration
-							15,	//WASC-15: Application Misconfiguration
-			    			msg); //HttpMessage
-			    		parent.raiseAlert(id, alert);
+			if (STSOption == null && msg.getRequestHeader().isSecure()) { // Header NOT found
+					raiseAlert(VULN_TYPE.HSTS_MISSING, msg, id);
+					return; //No point continuing
+			} else { 
+				for (String stsHeader : STSOption) {

```

Comment:
Maybe we should raise other issue if more than one header is being sent? (as per section 8.1 of rfx6797)

Also, should we alert if the header is malformed? (is the same thing as not sending it)